### PR TITLE
Make use MakeTriviaArray and AddRange.

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
Make use `MakeTriviaArray` (which caches some builders) and `AddRange` with in the scanner.

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb [hunk 1]
```diff
@@ -878,9 +878,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
                             Return
                         End If
 
-                        For i = 0 To triviaList.Count - 1
-                            tList.Add(triviaList(i))
-                        Next
+                        tlist.AddRange(MakeTriviaArray(triviaList))
```
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: this seems to be instantiating an intermediary collection now, just to deconstruct it.  that doesn't seem desirable.

Comment (AdamSpeight2008) [reply src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: The `MakeTriviaArray` is badly named as it doesn't return an array, but a `SyntaxList(Of VisualBasicSyntaxNode)`

Comment (CyrusNajmabadi) [reply src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: I realize that :)  But that SyntaxList causes allocations.  That seems highly undesirable in this codepath.

**Comment by jcouv** [comment]
Per comments above, the change introduces some allocations which doesn't seem worthwhile. 
I'll go ahead and close this PR, as it's getting old. Thanks

**Pull request closed without merging**